dipy/reconst/dki.py

```diff
@@ -0,0 +1,1606 @@
+        Axis of `evals` which contains 3 eigenvalues.
+
+    Returns
+    -------
+    rk : array
+        Calculated RK.
+
+    Notes
+    --------
+    RK is calculated with the following equation:
+
+    .. math::
+
+
+    K_{r}=G_1(\lambda_1,\lambda_2,\lambda_3)\hat{W}_{2222}+G_1(\lambda_1,\lambda_3,\lambda_2)\hat{W}_{333}+G_2(\lambda_1,\lambda_2,\lambda_3)\hat{W}_{2233}
+
+    where:
+    \begin{equation}
+    G_1(\lambda_1,\lambda_2,\lambda_3)=\frac{(\lambda_1+\lambda_2+\lambda_3)^2}{18\lambda_2(\lambda_2-\lambda_3)}[2\lambda_2+\frac{\lambda_3^2-3\lambda_2 \lambda_3}{\sqrt{\lambda_2\lambda_3}}]
+    \end{equation}
+
+    \begin{equation}
+    G_2(\lambda_1,\lambda_2,\lambda_3)=\frac{(\lambda_1+\lambda_2+\lambda_3)^2}{(\lambda_2-\lambda_3)^2}[\frac{\lambda_2+\lambda_3}{\sqrt{\lambda_2\lambda_3}}-2]
+    \end{equation}
+
+
+    """
+    [W_xxxx,W_yyyy,W_zzzz,W_xxyy,W_xxzz,W_yyzz]=[Wrotat[...,0],Wrotat[...,1],Wrotat[...,1],Wrotat[...,3],Wrotat[...,4],Wrotat[...,5]]
+
+    RadKurt=G1m(evals[...,0],evals[...,1],evals[...,2])*W_yyyy+G1m(evals[...,0],evals[...,2],evals[...,1])*W_zzzz+G2m(evals[...,0],evals[...,1],evals[...,2])*W_yyzz     
+    return RadKurt
+
+
+def apparent_kurtosis_coef(dki_params, sphere):
+    r"""
+    Calculate the apparent kurtosis coefficient (AKC) in each direction of a
+    sphere.
+
+    Parameters
+    ----------
+    dki_params : ndarray (..., 27)
+        All parameters estimated from the diffusion kurtosis model.
+        Parameters are ordered as follow:
+            1) Three diffusion tensor's eingenvalues
+            2) Three lines of the eigenvector matrix each containing the first,
+               second and third coordinates of the eigenvector
+            3) Fifteen elements of the kurtosis tensor
+
+    sphere : a Sphere class instance
+        The AKC will be calculated for each of the vertices in the sphere
+
+    Notes
+    -----
+    For each sphere direction with coordinates $(n_{1}, n_{2}, n_{3})$, the
+    calculation of AKC is done using formula:
+
+    .. math ::
+        AKC(n)=\frac{MD^{2}}{ADC(n)^{2}}\sum_{i=1}^{3}\sum_{j=1}^{3}
+        \sum_{k=1}^{3}\sum_{l=1}^{3}n_{i}n_{j}n_{k}n_{l}W_{ijkl}
+
+    where $W_{ijkl}$ are the elements of the kurtosis tensor, MD the mean
+    diffusivity and ADC the apparent diffusion coefficent computed as:
+
+    .. math ::
+        ADC(n)=\sum_{i=1}^{3}\sum_{j=1}^{3}n_{i}n_{j}D_{ij}
+
+    where $D_{ij}$ are the elements of the diffusion tensor.
+    """
+
+    # Flat parameters
+    outshape = dki_params.shape[:-1]
+    dki_params = dki_params.reshape((-1, dki_params.shape[-1]))
+
+    # Split data
+    evals, evecs, kt = split_dki_param(dki_params)
+
+    # Compute MD
+    MD = mean_diffusivity(evals)
+
+    # Initialize AKC matrix
+    V = sphere.vertices
+    AKC = np.zeros((len(kt), len(V)))
+
+    # loop over all voxels
+    for vox in range(len(kt)):
+        R = evecs[vox]
+        dt = lower_triangular(np.dot(np.dot(R, np.diag(evals[vox])), R.T))
```

Comment:
I believe there is a function in `reconst.dti` that computes the lower triangular elements from the evals/evecs that might be more efficient than this.